src/mesh/Utils.hpp

```diff
@@ -44,6 +44,27 @@ inline double edgeLength(const Edge &e)
   return (e.vertex(0).getCoords() - e.vertex(1).getCoords()).norm();
 }
 
+/** Calulates the area of an Triangle
+ *
+ * @param[in] e the triangle
+ *
+ * @returns surface area of the triangle. If the points are co-linear, the length of the edge.
+ */
+inline double triangleArea(const Triangle &t)
+{
+  Eigen::Vector3d vectorA = t.edge(1).vertex(1).getCoords() - t.edge(1).vertex(0).getCoords();
+  Eigen::Vector3d vectorB = t.edge(0).vertex(1).getCoords() - t.edge(0).vertex(0).getCoords();
+  // Compute cross-product of vector A and vector B
+  auto area = 0.5 * vectorA.cross(vectorB).norm();
+  // Check if the vectors are co-linear (Pseudo 3D case)
+  if (area < 1e-9) {
+    // Area is the longest edge.
+    std::vector<double> edgeLengths{edgeLength(t.edge(0)), edgeLength(t.edge(1)), edgeLength(t.edge(2))};
+    area = *std::max_element(edgeLengths.begin(), edgeLengths.end());
+  }
+  return area;
+}
```

Comment:
I would prefer a `double Triangle::area() const` instead of a non-member function as it is easier to visually digest.